lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -1882,7 +1882,9 @@ def _set_prometheus_data(self, event: Optional[RelationJoinedEvent] = None):
         # Add alert rules from file
         if self.path_to_own_alert_rules:
             alert_rules.add_path(self.path_to_own_alert_rules, recursive=True)
-        # Add generic alert rules
+        # Remove duplicated data and add generic alert rules
+        identifier = self.topology.identifier.replace("-", "_")
+        groups = [g for g in groups if not g["name"].startswith(identifier)]
         alert_rules.add(
```

Comment:
Here, it just adds alert_rule. so I put pruning before this. isn't it root cause of the duplication?

Reply:
@MichaelThamm Could you please check this comment if you have time? Thanks in advance.